web/app/Models/UserMessage.php

```diff
@@ -256,7 +256,7 @@ public function scopeStarred(Builder $query, User $user) : Builder
     public function scopeSent(Builder $query, User $user) : Builder
     {
         return $query->where('from_user_id', $user->id)
-            ->where('flags', '&', self::MESSAGE_SENT);
+            ->whereRaw('flags::int::bit(16) & ' . self::MESSAGE_SENT . '::bit(16) != 0::bit(16)');
```

Comment:
Given we have to repeat this ugly raw query fragment for each method that uses bitflags, and that we may use bitflags on other tables in the future, do you think it would make sense creating query helpers for these? Get, set, and clear. Like `->whereRaw(BitFlags::check('flags', self::MESSAGE_SENT))`?

Reply:
This is a good idea and I've added `flagIsSet(int $flag)` and `flagIsNotSet(int $flag)` to the `UsesBitmasks` trait to make this happen.